### PR TITLE
Add support for long keyID values

### PR DESCRIPTION
**Pull request by youennf**
Fixes https://github.com/w3c/webrtc-insertable-streams/issues/69

***
<a href="https://pr-preview.s3.amazonaws.com/youennf/webrtc-insertable-streams/pull/77.html" title="Last updated on Mar 19, 2021, 10:06 AM UTC (150a68d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-encoded-transform/77/b5fae15...youennf:150a68d.html" title="Last updated on Mar 19, 2021, 10:06 AM UTC (150a68d)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -218,11 +221,12 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
 ## Methods
 The <dfn method for="SFrameTransform">setEncryptionKey(|key|, |keyID|)</dfn> method steps are:
 1. Let |promise| be [=a new promise=].
-2.  [=In parallel=], run the following steps:
+2. If |keyID| is a <a href="https://heycam.github.io/webidl/#idl-bigint">bigint</a> which cannot be represented as a value in the range [0, 2<sup>64</sup>-1], [=reject=] |promise| with a {{RangeError}} exception.
```
Comment (youennf) [inline index.bs:224]: EnforceRange is throwing a TypeError, maybe we should do the same for BigInts that are too big.

**Comment by jan-ivar** [comment]
@martinthomson does this look good to you?

File: index.bs (hunk 1 above)
Comment (martinthomson) [inline index.bs:224]:
```suggestion
2. If |keyID| is a <a href="https://heycam.github.io/webidl/#idl-bigint">BigInt</a> which cannot be represented as a integer between 0 and 2<sup>64</sup>-1 inclusive, [=reject=] |promise| with a {{RangeError}} exception.
```

I don't know if the range notation will be understandable to all readers (if this is already an established convention, feel free to just take the CamelCase tweak.

**Review by martinthomson — APPROVED** [review]
Looks reasonable.  Assuming that SFrame is good (which might be in doubt).

File: index.bs [hunk 2]
```diff
@@ -218,11 +221,12 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
 ## Methods
 The <dfn method for="SFrameTransform">setEncryptionKey(|key|, |keyID|)</dfn> method steps are:
 1. Let |promise| be [=a new promise=].
-2.  [=In parallel=], run the following steps:
+2. If |keyID| is a <a href="https://heycam.github.io/webidl/#idl-bigint">BigInt</a> which cannot be represented as a integer between 0 and 2<sup>64</sup>-1 inclusive, [=reject=] |promise| with a {{RangeError}} exception.
```
Comment (youennf) [inline index.bs:224]:
```suggestion
2. If |keyID| is a <a href="https://heycam.github.io/webidl/#idl-bigint">bigint</a> which cannot be represented as a integer between 0 and 2<sup>64</sup>-1 inclusive, [=reject=] |promise| with a {{RangeError}} exception.
```

File: index.bs [hunk 3]
```diff
@@ -218,11 +221,12 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
 ## Methods
 The <dfn method for="SFrameTransform">setEncryptionKey(|key|, |keyID|)</dfn> method steps are:
 1. Let |promise| be [=a new promise=].
-2.  [=In parallel=], run the following steps:
+2. If |keyID| is a <a href="https://heycam.github.io/webidl/#idl-bigint">bigint</a> which cannot be represented as a integer between 0 and 2<sup>64</sup>-1 inclusive, [=reject=] |promise| with a {{RangeError}} exception.
```
Comment (youennf) [inline index.bs:224]:
```suggestion
2. If |keyID| is a <a href="https://heycam.github.io/webidl/#idl-bigint">BigInt</a> which cannot be represented as a integer between 0 and 2<sup>64</sup>-1 inclusive, [=reject=] |promise| with a {{RangeError}} exception.
```

**Comment by youennf** [comment]
@dontcallmedom any idea how to handle BigInt missing reference?

**Comment by dontcallmedom** [comment]
The right fix is to get bikeshed to recognize it, which itself require a fix in tis WebIDL parser - I've filed a pull request towards that: https://github.com/plinss/widlparser/pull/64

I think temporarily you can defined BigInt as a local anchor with type interface to workaround the problem while this gets fixed.

File: index.bs [hunk 4]
```diff
@@ -191,10 +191,16 @@ dictionary SFrameTransformOptions {
     SFrameTransformRole role = "encrypt";
 };
 
+// FIXME: Remove BigInt declaration once WebIDL parser is aware of it.
+interface BigInt {
+};
+typedef [EnforceRange] unsigned long long SmallCryptoKeyID;
+typedef (SmallCryptoKeyID or BigInt) CryptoKeyID;
+
```
Comment (dontcallmedom) [inline index.bs:199]: What I had meant was instead to add in the [bikeshed metadata](https://tabatkins.github.io/bikeshed/#custom-dfns)
```html
<pre class=anchors>
url: https://heycam.github.io/webidl/#idl-bigint; type: interface; text: BigInt
</pre>```

Comment (youennf) [reply index.bs:199]: Ah, thanks, updated the PR accordingly

**Comment by youennf** [comment]
This PR was discussed during Thursday editor's call and ready to merge once the WebIDL BigInt issue is fixed.

**Comment by youennf** [comment]
Checks are fine, so merging.

File: index.bs [hunk 5]
```diff
@@ -191,10 +194,13 @@ dictionary SFrameTransformOptions {
     SFrameTransformRole role = "encrypt";
 };
 
+typedef [EnforceRange] unsigned long long SmallCryptoKeyID;
+typedef (SmallCryptoKeyID or BigInt) CryptoKeyID;
```
Comment (foolip) [inline index.bs:198]: I've sent https://github.com/w3c/webrtc-encoded-transform/pull/92 to fix the case of `BigInt` here, spotted by tests in https://github.com/w3c/webref.

**Pull request merged**